File: frontend/src/model/Album.model.ts
import { Dayjs } from 'dayjs';

import { IImageLink } from '@/model/ImageLink.model';
import { IImage } from '@/model/Image.model';

export interface IAlbum {
    id: string;
    title: string;
    imageCount: number;
    link: string;
    createdAt: Dayjs;
    coverImage: IImageLink;
    images: Array<IImage> | null;
}

File: backend/Models/AlbumModel.cs
namespace Models;

public sealed class AlbumModel
{
    public string Id { get; init; } = null!;
    public string Title { get; init; } = null!;
    public int ImageCount { get; init; }
    public string Link { get; init; } = null!;
    public DateTime CreatedAt { get; init; }
    public ImageLinkModel CoverImage { get; init; } = null!;
    public List<ImageModel>? Images { get; init; }
}